packages/compass-aggregations/src/utils/get-schema.ts

```diff
@@ -0,0 +1,42 @@
+import type { Document } from 'mongodb';
+
+const getArrayKeys = (records: Document[]) => {
+  return records
+    .map((item) => getObjectKeys(item))
+    .flat()
+    .filter((x, i, a) => a.indexOf(x) === i)
+    .sort();
+};
+
+const getObjectKeys = (record: Document) => {
+  const keys: string[] = [];
+
+  if (!record) {
+    return keys;
+  }
+
+  for (const key in record) {
+    keys.push(key);
+    const value = record[key];
+
+    if (value && typeof value === 'object') {
+      const isBson = value._bsontype;
+      if (isBson) {
+        keys.push(key);
+      } else {
+        const nestedKeys = Array.isArray(value)
+          ? getArrayKeys(value)
+          : getObjectKeys(value);
+        nestedKeys.forEach((nestedKey) => {
+          keys.push(`${key}.${nestedKey}`);
+        });
+      }
```

Comment:
```suggestion
      if (!isBson) {
        const nestedKeys = Array.isArray(value)
          ? getArrayKeys(value)
          : getObjectKeys(value);
        nestedKeys.forEach((nestedKey) => {
          keys.push(`${key}.${nestedKey}`);
        });
      }
```
Since you already push the same key on Line - 19, we probably can avoid inserting duplicate keys this way? then we won't have to do the `filter` operation in the `getArrayKeys` method.